pkg/scheduler/api/compatibility/compatibility_test.go

```diff
@@ -1218,34 +1224,48 @@ func TestCompatibility_v1_Scheduler(t *testing.T) {
 		if !reflect.DeepEqual(policy, tc.ExpectedPolicy) {
 			t.Errorf("%s: Expected:\n\t%#v\nGot:\n\t%#v", v, tc.ExpectedPolicy, policy)
 		}
-
+		policyConfigMap := v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{Namespace: metav1.NamespaceSystem, Name: "scheduler-custom-policy-config"},
+			Data:       map[string]string{schedulerconfig.SchedulerPolicyConfigMapKey: tc.JSON},
+		}
+		policyConfigMap.APIVersion = "v1"
+		body, err := json.Marshal(policyConfigMap)
+		if err != nil {
+			t.Errorf("Error creating configMap: %+v", err)
+		}
 		handler := utiltesting.FakeHandler{
 			StatusCode:   500,
-			ResponseBody: "",
+			ResponseBody: string(body),
 			T:            t,
 		}
 		server := httptest.NewServer(&handler)
 		defer server.Close()
 		client := clientset.NewForConfigOrDie(&restclient.Config{Host: server.URL, ContentConfig: restclient.ContentConfig{GroupVersion: &schema.GroupVersion{Group: "", Version: "v1"}}})
+		algorithmSrc := schedulerconfig.SchedulerAlgorithmSource{
+			Policy: &schedulerconfig.SchedulerPolicySource{},
+		}
 		informerFactory := informers.NewSharedInformerFactory(client, 0)
 
-		if _, err := factory.NewConfigFactory(&factory.ConfigFactoryArgs{
-			Client:                         client,
-			NodeInformer:                   informerFactory.Core().V1().Nodes(),
-			PodInformer:                    informerFactory.Core().V1().Pods(),
-			PvInformer:                     informerFactory.Core().V1().PersistentVolumes(),
-			PvcInformer:                    informerFactory.Core().V1().PersistentVolumeClaims(),
-			ReplicationControllerInformer:  informerFactory.Core().V1().ReplicationControllers(),
-			ReplicaSetInformer:             informerFactory.Apps().V1().ReplicaSets(),
-			StatefulSetInformer:            informerFactory.Apps().V1().StatefulSets(),
-			ServiceInformer:                informerFactory.Core().V1().Services(),
-			PdbInformer:                    informerFactory.Policy().V1beta1().PodDisruptionBudgets(),
-			StorageClassInformer:           informerFactory.Storage().V1().StorageClasses(),
-			CSINodeInformer:                informerFactory.Storage().V1beta1().CSINodes(),
-			HardPodAffinitySymmetricWeight: v1.DefaultHardPodAffinitySymmetricWeight,
```

Comment:
+1 for removing `HardPodAffinitySymmetricWeight`, `DisablePreemption` and `PercentageOfNodesToScore`. Function `scheduler.New` defaults to `v1.DefaultHardPodAffinitySymmetricWeight`, `false` and `schedulerapi.DefaultPercentageOfNodesToScore` when those settings are not passed through `opts ...func(o *schedulerOptions)`.